### PR TITLE
docs: Update compatibility.md

### PR DESCRIPTION
**Pull request by Dimrok**
**- What I did**

- I  merged the two matrices to get a clearer view.
- I added docker-compose
- I added version 2.x

**- Description for the changelog**

Update documentation regarding composefile version compatibility.

**- A picture of a cute animal (not mandatory but encouraged)**

image
(credits: u/kaitlynsara06 from reddit)

### CODE REVIEW

File: docs/compatibility.md [hunk 1]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
+|     - parallelism            |  S  |     |     |     |     |     |     |     |     |     |     |     |     |
+|   - update_config            |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - delay                  |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - failure_action         |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - max_failure_ratio      |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - monitor                |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - order                  |  SK |  SK |  SK |  SK |     |     |     |     |     |     |     |     |     |
+|     - parallelism            |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+| - devices                    |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - depends_on                 |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
```
Comment (vdemeester) [inline docs/compatibility.md]: Doesn't work on `swarm`, there is no notion of `depends_on` in swarm, the key is ignored (should be in `UnsupportedProperties`, not sur why it's not…)

File: docs/compatibility.md [hunk 2]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
+| - healthcheck                |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |     |
+|   - disable                  |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
+|   - interval                 |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - retries                  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - start_period             |  CS |  CS |  CS |  CS |     |     |     |     |
+|   - test                     |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - timeout                  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - hostname                   |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - image                      |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - init                       |  *  |     |     |     |     |     |     |     |     |     |     |     |     |
```
Comment (vdemeester) [inline docs/compatibility.md]: Probably not support in Kubernetes if it's in 3.7, might be wrong though

File: docs/compatibility.md [hunk 3]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
+| - ipc                        |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |
+| - isolation                  |  CS |  CS |  CS |     |     |     |     |     |     |     |     |     |     |
+| - labels                     |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - links                      |     |     |     |     |     |     |     |     |     |     |     |     |     |
```
Comment (vdemeester) [inline docs/compatibility.md]: Should be supported in `C` (compose) I think

File: docs/compatibility.md [hunk 4]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
-| - shm_size                   |     |     |     |     |     |     |     |
-| - stdin_open                 |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| - tty                        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| - user                       |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-|   -- numerical               |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-|   -- name                    |     |     |     |     |     |     |     |
-| - working_dir                |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| Features                     | 3.7 | 3.6 | 3.5 | 3.4 | 3.3 | 3.2 | 3.1 | 3.0 | 2.4 | 2.3 | 2.2 | 2.1 | 2.0 |
+|------------------------------|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|
+| configs                      |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
```
Comment (vdemeester) [inline docs/compatibility.md]: Config should work the same way `secret` are…

File: docs/compatibility.md [hunk 5]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
+| - networks                   |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
+|   - aliases                  |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
+|   - ipv4_address             |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
+|   - ipv6_address             |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |
+|   - name                     |  CS |  CS |  CS |  CS |     |     |     |     |     |     |     |     |     |
+| - pid                        |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - ports                      |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - mode:                    |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|     - host                   | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? |
+|     - ingress                | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? | CS? |
```
Comment (vdemeester) [inline docs/compatibility.md]: `ingress` is swarm only

File: docs/compatibility.md [hunk 6]
```diff
@@ -2,262 +2,161 @@ List of supported features, depending on the Docker Compose version.
...
+| - userns_mode                |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - volumes (long syntax)      |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|   - read_only                |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|   - source                   |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|   - target                   |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|   - type                     |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|     - volume                 |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|     - bind                   |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|     - tmpfs                  |  *  |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |
+|   - bind                     |  CS |  CS |  CS |  CS |  CS |  CS |     |     |     |     |     |     |     |
```
Comment (vdemeester) [inline docs/compatibility.md]: Didn't we had support for bind mount in k8s ? :thinking:

Comment (vdemeester) [reply docs/compatibility.md]: cc @simonferquel

File: docs/compatibility.md [hunk 7]
```diff
@@ -1,263 +1,187 @@
...
+|     - host                   | CSk | CSk | CSk | CSk | CSk | CSk | CSk | CSk |  CS |  CS |  CS |  CS |  CS |
+|     - ingress                | CSk | CSk | CSk | CSk | CSk | CSk | CSk | CSk |  CS |  CS |  CS |  CS |  CS |
+|   - protocol                 |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - published                |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+|   - target                   |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - privileged                 |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  CK |
+| - read_only                  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - restart                    |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - runtime                    |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |  CS |     |     |     |
+| - secrets                    |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
```
Comment (shin-) [inline docs/compatibility.md]: For secrets, `uid`, `gid` and `mode` are not support for Compose. Additionally, Compose does not support secrets declared external.

File: docs/compatibility.md [hunk 8]
```diff
@@ -1,263 +1,187 @@
...
+| - cap_add                    |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cap_drop                   |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cpu_shares                 |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - cpu_quota                  |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - cpuset                     |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - command                    |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - configs                    |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
+| - cgroup_parent              |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - container_name             |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - credential_spec            |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
```
Comment (shin-) [inline docs/compatibility.md]: Note that support for `credential_spec` in Compose is pending: https://github.com/docker/compose/issues/5488

File: docs/compatibility.md [hunk 9]
```diff
@@ -1,263 +1,187 @@
...
+|   - file                     |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
+|   - registry                 |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
+| - deploy                     |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - endpoint_mode            |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|     - dnsrr                  |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|     - vip                    |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|   - labels                   |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - mode                     |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - placement                |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - replicas                 |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
```
Comment (shin-) [inline docs/compatibility.md]: Supported in Compose if *compatibility mode* is enabled: https://docs.docker.com/compose/compose-file/compose-versioning/#compatibility-mode

File: docs/compatibility.md [hunk 10]
```diff
@@ -1,263 +1,187 @@
...
+|   - resources                |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - limits                 |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|     - reservations           |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - restart_policy           |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
```
Comment (shin-) [inline docs/compatibility.md]: Partially supported in Compose if *compatibility mode* is enabled: https://docs.docker.com/compose/compose-file/compose-versioning/#compatibility-mode

File: docs/compatibility.md [hunk 11]
```diff
@@ -1,263 +1,187 @@
...
+|   - registry                 |  *  |  *  |  *  |  *  |  *  |     |     |     |     |     |     |     |     |
+| - deploy                     |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - endpoint_mode            |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|     - dnsrr                  |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|     - vip                    |  S  |  S  |  S  |  S  |  S  |     |     |     |     |     |     |     |     |
+|   - labels                   |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - mode                     |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - placement                |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - replicas                 |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
+|   - resources                |  SK |  SK |  SK |  SK |  SK |  SK |  SK |  SK |     |     |     |     |     |
```
Comment (shin-) [inline docs/compatibility.md]: Supported in Compose if *compatibility mode* is enabled: https://docs.docker.com/compose/compose-file/compose-versioning/#compatibility-mode

File: docs/compatibility.md (hunk 1 above)
Comment (Dimrok) [reply docs/compatibility.md]: The depends_on option is ignored when deploying a stack in swarm mode with a version 3 Compose file. 

Good catch.

File: docs/compatibility.md (hunk 2 above)
Comment (Dimrok) [reply docs/compatibility.md]: Fixed.

File: docs/compatibility.md (hunk 7 above)
Comment (Dimrok) [reply docs/compatibility.md]: Fixed.

File: docs/compatibility.md (hunk 10 above)
Comment (Dimrok) [reply docs/compatibility.md]: Fixed.

File: docs/compatibility.md (hunk 9 above)
Comment (Dimrok) [reply docs/compatibility.md]: Fixed.

File: docs/compatibility.md (hunk 11 above)
Comment (Dimrok) [reply docs/compatibility.md]: Fixed.

File: docs/compatibility.md [hunk 12]
```diff
@@ -1,263 +1,192 @@
...
+|   - args                     |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - cache_from               |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - context                  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - dockerfile               |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - extra_hosts              |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - labels                   |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - network                  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - shm_size                 |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - target                   |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cap_add                    |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  C  |  C  |  C  |  C  |  C  |  C  |
```
[the remark this answers is not in the document]
Comment (simonferquel) [reply docs/compatibility.md]: Just re-checked they are not, just wasn't looking at the right place

**Comment by chris-crone** [comment]
Ping @vdemeester and @simonferquel to update their reviews

File: docs/compatibility.md [hunk 13]
```diff
@@ -1,263 +1,192 @@
...
-| - shm_size                   |     |     |     |     |     |     |     |
-| - stdin_open                 |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| - tty                        |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-| - user                       |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-|   -- numerical               |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
-|   -- name                    |     |     |     |     |     |     |     |
-| - working_dir                |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
+| Features                     | 3.7 | 3.6 | 3.5 | 3.4 | 3.3 | 3.2 | 3.1 | 3.0 | 2.4 | 2.3 | 2.2 | 2.1 | 2.0 |
+|------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| configs.<name> (long syntax) |  CK |  CK |  CK |  CK |  CK |     |     |     |     |     |     |     |     |
```
Comment (shin-) [inline docs/compatibility.md]: The whole configs section should be `SK`, not `CK`

**Comment by chris-crone** [comment]
Ping @shin- for a review update

File: docs/compatibility.md [hunk 14]
```diff
@@ -1,263 +1,192 @@
...
+|   - network                  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - shm_size                 |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+|   - target                   |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cap_add                    |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cap_drop                   |  CK |  CK |  CK |  CK |  CK |  CK |  CK |  C  |  C  |  C  |  C  |  C  |  C  |
+| - cpu_shares                 |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - cpu_quota                  |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - cpuset                     |     |     |     |     |     |     |     |     |  C  |  C  |  C  |  C  |  C  |
+| - command                    |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |  *  |
+| - configs                    |  CK |  CK |  CK |  CK |  CK |     |     |     |     |     |     |     |     |
```
[the remark this answers is not in the document]
Comment (chris-crone) [reply docs/compatibility.md]: Since @Dimrok is on PTO, can we merge and follow up?

**Comment by ijc** [comment]
This seems like a pretty massive/intractable PR review-wise, given that it clearly contains an improvement over the status quo perhaps we should merge it and correct that last few remaining inaccuracies in a followup (or followups)?

File: docs/compatibility.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (shin-) [reply docs/compatibility.md]: Yep, intentionally left as a comment as to not block this any further. Merge away!

[the remark this answers is not in the document]
Comment (shin-) [reply docs/compatibility.md]: Yep, intentionally left as a comment as to not block this any further. Merge away!

**Pull request merged**